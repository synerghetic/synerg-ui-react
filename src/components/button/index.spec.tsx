import React from 'react'
import Button from '.'
import { mount } from 'enzyme'

describe('Button', () => {
    it('should render', () => {
        const component = mount(<Button>Hello World</Button>)

        expect(component.html()).toContain('button')
        expect(component.html()).toMatchSnapshot()
    })

    it('should have the primary class', () => {
        const component = mount(<Button>Hello World</Button>)

        expect(component.find('button')).toHaveClassName('button--primary')
    })

    it('should call onClick one time', () => {
        const onClick = jest.fn()

        const component = mount(<Button onClick={onClick}>Hello World</Button>)

        component.simulate('click')

        expect(onClick.mock.calls.length).toBe(1)
    })
})
