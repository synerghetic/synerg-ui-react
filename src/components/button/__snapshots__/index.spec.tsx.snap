// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `"<button class=\\"button--primary  \\"><div class=\\"loader-container hidden\\"><div class=\\"loader\\" style=\\"width: 12px; height: 12px; border-top: 4px solid #fff; border-right: 4px solid #59089e; border-left: 4px solid #59089e; border-bottom: 4px solid #59089e;\\"></div></div><span class=\\"\\">Hello World</span></button>"`;
